File: rsleetcode/src/easy/palindrome_number.rs
// 9. Palindrome Number
//
// https://leetcode.com/problems/palindrome-number/


#[allow(dead_code)]
pub struct Solution {
}

#[allow(dead_code)]
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let x: String = x.to_string();
        let reversed: String = x.chars().rev().collect();
        x == reversed
    }
}

#[allow(dead_code)]
pub fn run() {
    debug_assert!(Solution::is_palindrome(121));
    debug_assert!(!Solution::is_palindrome(-121));
    debug_assert!(!Solution::is_palindrome(10));
}

File: easy/palindrome_number.py
"""9. Palindrome Number.

https://leetcode.com/problems/palindrome-number/
"""


class Solution:

    def isPalindrome(self, x: int) -> bool:  # noqa
        str_x = str(x)
        return str_x == str_x[::-1]


def test():
    assert Solution().isPalindrome(121)
    assert not Solution().isPalindrome(-121)
    assert not Solution().isPalindrome(10)
